src/pages/ct-poe-groups/forms/PurposeOfExchangeGroupForm.js

```diff
@@ -0,0 +1,104 @@
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useForm } from 'src/hooks/form'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { CurrencyTradingSettingsRepository } from 'src/repositories/CurrencyTradingSettingsRepository'
+
+export default function PurposeOfExchangeGroupForm({ labels, recordId, maxAccess }) {
+  const { platformLabels } = useContext(ControlContext)
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: CurrencyTradingSettingsRepository.PurposeExchangeGroup.page
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      name: '',
+      flName: '',
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      name: yup.string().required(),
+    }),
+    onSubmit: async obj => {
+      try {
+        const response = await postRequest({
+          extension: CurrencyTradingSettingsRepository.PurposeExchangeGroup.set,
+          record: JSON.stringify(obj)
+        })
+
+        if (!obj.recordId) {
+          toast.success(platformLabels.Added)
+          formik.setFieldValue('recordId', response.recordId)
+        } else toast.success(platformLabels.Edited)
+        invalidate()
+      } catch (error) {}
+    }
+  })
+
+  const editMode = !!formik.values.recordId
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (recordId) {
+          const res = await getRequest({
+            extension: CurrencyTradingSettingsRepository.PurposeExchangeGroup.get,
+            parameters: `_recordId=${recordId}`
+          })
+
+          formik.setValues(res.record)
+        }
+      } catch (error) {}
+    })()
+  }, [])
+
+  return (
+    <FormShell resourceId={ResourceIds.PurposeExchangeGroup} form={formik} maxAccess={maxAccess} editMode={editMode}>
+      <VertLayout>
+        <Grow>
+          <Grid container spacing={4}>
+            <Grid item xs={12}>
+              <CustomTextField
+                name='name'
+                label={labels.name}
+                value={formik.values.name}
+                required
+                maxLength='30'
+                maxAccess={maxAccess}
+                onChange={formik.handleChange}
+                onClear={() => formik.setFieldValue('name', '')}
+                error={formik.touched.name && formik.errors.name}
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <CustomTextField
+                name='flName'
+                label={labels.flName}
+                value={formik.values.flName}
+                maxAccess={maxAccess}
+                maxLength='40'
```

Comment:
Are u sure about maximum length? I didn't see it mentioned in the task description

Reply:
no need for it